packages/js/src/event-emitter/types.ts

```diff
@@ -1,11 +1,16 @@
-import { INotificationDto } from '@novu/shared';
+import { ISessionDto } from '@novu/shared';
+import { Notification } from '../feeds/notification';
 
 import { PaginatedResponse } from '../types';
 
 export type Events = {
+  'session.initialize.*': undefined;
+  'session.initialize.pending': undefined;
```

Comment:
I am torn about using separate events for all the loading state vs one event for the output of the fetch with a { data, error } payload. 

I suggest we have a mini DX guide on how a developer can use them and see what’s better. For now we can leave them as such.

Reply:
the events I've improved in the next PR: https://github.com/novuhq/novu/pull/5688